src/server/helpers/collectionRouteUtils.js

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2020 Prabal Singh
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import * as handler from '../helpers/handler';
+import * as search from './search';
+import _ from 'lodash';
+
+
+export async function makeCollectionCreateOrEditHandler(req, res) {
+	try {
+		const {UserCollection, UserCollectionCollaborator} = req.app.local.orm;
+		const isNew = !res.locals.collection;
+		let newCollection;
+		let method;
+		if (isNew) {
+			newCollection = await new UserCollection({
+				ownerId: req.user.id
+			});
+			method = 'insert';
+		}
+		else {
+			newCollection = await new UserCollection({id: req.params.collectionId}).fetch({
+				require: true
+			});
+			method = 'update';
```

Comment:
In this else block I guess you should also set lastModified with the current timestamp, otherwise it is not updated.

Reply:
I see you commented out the last_modified update below. What's the reason?